edx-solutions/xblock-drag-and-drop-v2
# [MCKIN-8133] Improve i18n tooling

This PR improves upon PR #156 by adding support for generating dummy translations using `edx-i18n-tools`. It also includes make targets to automate extracting, building dummy translations and compiling.

Sandbox
---------

https://pr137.sandbox.stage.opencraft.hosting/
https://studio-pr137.sandbox.stage.opencraft.hosting/

Testing
-------

1. Run `make extract_translations` to extract Python, HTML and JS strings.
2. Make sure all strings were correctly extracted to `django.po` and `djangojs.po`.
3. Remove generated files for translations.
4. Run `validate_translations` to compile, build dummy and detect changes.
5. Make sure it results in `Source translation files are current.`
6. Install the xblock in a devstack, add it to a Unit in studio.
7. Change the language to `eo` using the `/update_lang` path.
8. Confirm the strings are translated, both from HTML and JavaScript.

Expected results
-----------------
![screen shot 2018-08-21 at 17 05 12](https://user-images.githubusercontent.com/5691347/44428565-207dd000-a56b-11e8-8d90-8eb66716184a.png)
![screen shot 2018-08-21 at 17 05 33](https://user-images.githubusercontent.com/5691347/44428568-25428400-a56b-11e8-94e1-7a92b74b60a6.png)
![screen shot 2018-08-21 at 17 06 58](https://user-images.githubusercontent.com/5691347/44428574-2a073800-a56b-11e8-8727-60f28131d208.png)

Reviewers
----------
- [ ] @ahmedaljazzar
